### DISCUSSION TITLE
Ubuntu 24.04, building with make command fails

### DISCUSSION THREAD

**Discussion by tricarte** [vlang/v#21880] [Installation Issues]
Followed [the official building instructions](https://github.com/vlang/v#installing-v-from-source). But `make` command gives below error.

```
make fresh_vc
make[1]: Entering directory '/home/username/repos/v'
rm -rf ./vc
git clone --filter=blob:none --quiet https://github.com/vlang/vc ./vc
make[1]: git: Permission denied
make[1]: *** [GNUmakefile:142: fresh_vc] Error 127
make[1]: Leaving directory '/home/username/repos/v'
make: *** [GNUmakefile:191: vc/.git/config] Error 2
```

If I manually clone vc into v directory with: `git clone --filter=blob:none https://github.com/vlang/vc vc/`

then the `make` command fails this way:

```
cd ./vc && git clean -xf && git pull --quiet
make fresh_tcc
make[1]: Entering directory '/home/username/repos/v'
rm -rf ./thirdparty/tcc
git clone --filter=blob:none --quiet --branch thirdparty-linux-amd64 https://github.com/vlang/tccbin ./thirdparty/tcc
make[1]: git: Permission denied
make[1]: *** [GNUmakefile:156: fresh_tcc] Error 127
make[1]: Leaving directory '/home/username/repos/v'
make: *** [GNUmakefile:188: thirdparty/tcc/.git/config] Error 2
```

Ok let's clone that tccbin manually: `git clone --filter=blob:none --quiet --branch thirdparty-linux-amd64 https://github.com/vlang/tccbin ./thirdparty/tcc`

and then run `make` again

```
cd ./vc && git clean -xf && git pull --quiet
cd ./thirdparty/tcc && git clean -xf && git pull --quiet
cc  -std=gnu99 -w -o v1.exe ./vc/v.c -lm -lpthread 
./v1.exe -no-parallel -o v2.exe  cmd/v
./v2.exe -nocache -o ./v  cmd/v
rm -rf v1.exe v2.exe
Your `tcc` is working. Good - it is much faster at compiling C source code.
V has been successfully built
V 0.4.6 615a9a0
```

Now compiled successfuly.

**Comment by spytheman** [comment]
Imho something is wrong with your PATH or git installations.

Can you try `ls -la $(which git)` and `git version` in a new shell?

**Comment by spytheman** [comment]
Also please check the permissions and owner info, for the `vc/` folder with `ls -lad vc`, right after cloning and `cd v`, before running `make`, and after running `make`.

**Comment by tricarte** [comment]
`ls -la $(which git)`:

```
-rwxr-xr-x 1 root root 4066200 May 20 14:15 /usr/bin/git
```

Git version: 2.43.0.

OK, I'm starting again from the beginning: There is no `vc` directory inside the source root after cloning v. Also the `vc` directory does not appear after issuing the `make` command as it fails giving the first `permission denied` error I mentioned earlier.

After manually cloning `vc` with: `git clone --filter=blob:none https://github.com/vlang/vc vc/`:

`ls -lad vc`:

`drwxrwxr-x 3 my_user my_user 61 Jul 16 09:09 vc`

Running `make` again, still getting `git: Permission denied` error for `tcc` this time.

After manually cloning `tcc` with: `git clone --filter=blob:none --quiet --branch thirdparty-linux-amd64 https://github.com/vlang/tccbin ./thirdparty/tcc`

`ls -lad thirdparty/tcc/`:

`drwxrwxr-x 6 my_user my_user 105 Jul 18 14:01 thirdparty/tcc/`

So, manual git clone works fine. As you said, if no other one has this issue, it's probably due to my installation but I can't figure out why.
